File: Data Structures & Algorithms/two-integer-sum/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {

        HashMap<Integer, Integer> map = new HashMap<>();
        // {value : index}

        for (int i = 0; i < nums.length; i++) {
            int n = nums[i];
            Integer val = map.get(target - n);
            if (val != null) return new int[] {val, i};
            map.put(n, i);
        }

        return null;
        
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # use two pointer approach
        # guaranteed valid solution

        i: int = 0
        j: int = len(numbers) - 1

        while True:
            
            combine: int = numbers[i] + numbers[j]

            if combine > target:
                j -= 1
            elif combine < target: 
                i += 1
            else:
                break

        return list([i+1, j+1])